ksql-engine/src/test/resources/query-validation-tests/README.md

```diff
@@ -5,9 +5,191 @@ file contains one or more tests. They are written in a simple json format that d
+| properties       | (Optional) A map of property name to value. Can contain any valid Ksql config. The config is passed to the engine when executing the statements in the test case |
+| topics           | (Optional) An array of the topics this test case needs. Allows more information about the topic to be supplied, e.g. an existing Avro schema (See below for more info) |
+| inputs           | (Required if `expectedException` not supplied) The set of input messages to be produced to Kafka topic(s), (See below for more info) |
+| outputs          | (Required if `expectedException` not supplied) The set of output messages expected in the output topic(s), (See below for more info) |
+| expectedException| (Required in `inputs` and `outputs` not supplied) The exception that should be thrown when executing the supplied statements, (See below for more info) |
+
+### Formats
+A test case can optionally supply an array of formats to run the test case as.
+The current format will be injected into the statements of the test case as the `{FORMAT}` variable.
+
+For example:
+```json
+   {
+      "name": "my first test that will run with different formats",
+      "format": ["AVRO", "JSON"],
+      "statements": [
+        "CREATE TABLE TEST (ID bigint) WITH (kafka_topic='test_topic', value_format='{FORMAT}');",
+        ...
+      ],
+      ...
+```
+
+The test will run once for each defined format.
+
+Make use of the `format` option only where the serialized format of the data may affect the functionality.
+
+### Statements
+You can specify multiple statements per test case, i.e. to set up the various streams needed
+for joins etc, but currently only the final topology will be verified. This should be enough
 for most tests as we can simulate the outputs from previous stages into the final stage. If we
-take a modular approach to testing we can still verify that it all works correctly, i.e, if we,
-verify the output of a select or aggregate is correct, we can use simulated output to feed into,
-a join or another aggregate.
\ No newline at end of file
+take a modular approach to testing we can still verify that it all works correctly, i.e. if we
+verify the output of a select or aggregate is correct, then we can use simulated output to feed
+into a join or another aggregate.
+
+### Topics
+A test case can optionally supply an array of topics that should exist and additional information about those topics.
+It is not necessary to add entries for topics required by the test case unless any of the following attributes need to be controlled:
+(Tests will create input and output topics as required).
+
+| Attribute | Description |
+|-----------|:------------|
+| name      | (Required) the name of the topic |
+| schema    | (Optional) the schema, registered in the schema store, of the topic. |
+| format    | (Optional) the serialization format of records within the topic, e.g. `AVRO` |
```

Comment:
are there defaults for these Optional values?

Reply:
updated.